LanguageFeatures/Patterns/logical_or_A04_t01.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion logicalOrPattern ::= ( logicalOrPattern '|' )? logicalAndPattern
+///
+/// A pair of patterns separated by | matches if either of the branches match
+///
+/// A logical-or pattern may match even if one of its branches does not. That
+/// means that any variables in the non-matching branch would not be
+/// initialized. To avoid problems stemming from that, the following
+/// restrictions apply:
+///
+/// The two branches must define the same set of variables.
+///
+/// If the left branch matches, the right branch is not evaluated. This
+/// determines which value the variable gets if both branches would have
+/// matched. In that case, it will always be the value from the left branch.
+///
+/// @description Checks that if the left branch matches, the right branch is not
+/// evaluated
```

Comment:
I can't immediately see any reason why this test would detect the bug which is that the right branch _is_ evaluated in a situation where the left branch matches. Does it actually detect that situation?

You could say that we get the expected value for `s`, but that's not a safe test: The implementation could still evaluate both branches and then initialize `s` to the value that corresponds to the match that succeeded. Or it could evaluate the right operand first and then the left operand, or it could use some other ordering.

I think it's necessary to (1) have a side effect in a getter such that we can verify that a specific getter was indeed invoked, and (2) check that the getters corresponding to the patterns that should be skipped are not invoked.

An approach that I've seen in many language tests is like this:

```dart
String log = '';
void addToLog(String s) => log = log.isEmpty ? s : '$log, $s';

class C {
  int get g {
    addToLog('C.g');
    return 0;
  }
}
```

Reply:
The idea was that if the right branch is evaluated then `Point.area` getter will throw an exception. But now I see that it was a wrong idea. If the right branch is evaluated then an extractor name will be checked first. Runtime type of the variable is not a `Point`, so `Point.area` getter wont't be called.  I'll change the test in the way you've suggested. Thanx!